garage/envs/base.py

```diff
@@ -1,17 +1,156 @@
+        Parameterized.__init__(self)
+        super(GarageEnv, self).__init__(env)
+
+    @overrides
+    def action_space(self):
+        """
+        Classes inheriting from GarageEnv need to convert
+        action_space from gym.space to garage.space.
+
+        Returns:
+            NotImplementedError
+        """
+        raise NotImplementedError
+
+    @overrides
+    def observation_space(self):
+        """
+        Classes inheriting from GarageEnv need to convert
+        observation_space from gym.space to garage.space.
+
+        Returns:
+            NotImplementedError
+        """
+        raise NotImplementedError
+
+    def close(self):
+        """
+        Close the wrapped env.
+
+        Returns:
+            None
+        """
+        self.env.close()
+
+    def get_params_internal(self, **tags):
+        """
+        Returns an empty list if env.get_params() is called.
+
+        Returns:
+            An empty list
+        """
+        return []
+
+    @property
+    def horizon(self):
+        """
+        Get the maximum episode steps for the wrapped env.
+
+        Returns:
+            max_episode_steps (int)
+        """
+        if self.env.spec is not None:
+            return self.env.spec.max_episode_steps
+        else:
+            return NotImplementedError
+
+    def log_diagnostics(self, paths, *args, **kwargs):
+        pass
+
+    def render(self, *args, **kwargs):
+        return self.env.render(*args, **kwargs)
+
+    def reset(self, **kwargs):
+        return self.env.reset(**kwargs)
+
+    @property
+    def spec(self):
+        """
+        Returns an EnvSpec.
+
+        Returns:
+            spec (garage.envs.EnvSpec)
+        """
+        return EnvSpec(
+            observation_space=self.observation_space,
+            action_space=self.action_space,
+        )
+
+    def step(self, action):
+        return self.env.step(action)
+
+    def wrapped_env(self):
+        """
+        Returns the original env wrapped by this class.
+
+        Returns:
+            env (gym.Env)
+        """
+        return self.env
+
+    def _to_garage_space(self, space):
+        """
+        Converts a gym space into a garage space.
+
+        Args:
+            space (gym.spaces)
+
+        Returns:
+            space (garage.spaces)
+        """
+        if isinstance(space, GymBox):
+            return Box(low=space.low, high=space.high)
+        elif isinstance(space, GymDiscrete):
+            return Discrete(space.n)
+        elif isinstance(space, GymTuple):
+            return Product(
+                [self._to_garage_space(space) for space in space.spaces])
+        else:
+            raise NotImplementedError
 
-from garage.envs import EnvSpec
 
 _Step = collections.namedtuple("Step",
                                ["observation", "reward", "done", "info"])
 
 
-def Step(observation, reward, done, **kwargs):
+def Step(observation, reward, done, **kwargs):  # noqa: N802
```

Comment:
Hmm why not just `Step = namedtuple(...)` above? Seems like directly out of the documentation.

https://docs.python.org/3.6/library/collections.html#collections.namedtuple

Reply:
After removing `def Step(...)`, calling `Step()` in garage requires passing `info` as an explicit positional argument:
```
  File "/Users/jonathon/Documents/garage/garage/garage/envs/box2d/box2d_env.py", line 191, in step
    return Step(observation=next_obs, reward=reward, done=done)
TypeError: __new__() missing 1 required positional argument: 'info'
```
I believe it's better to leave it as-is and write a comment above `def Step` about its necessity.

Reply:
Ah I see. SGTM.